TemplateManagement/PowerShell/ActiveDirectoryJoinV2/Set-RenameValuesToSecretStore-TemplateVM.ps1

```diff
@@ -0,0 +1,102 @@
+#>
+
+
+[CmdletBinding()]
+param(
+    [Parameter(Mandatory,
+    ValueFromPipeline=$true, 
+    ValueFromPipelineByPropertyName=$true,
+    HelpMessage="Lab prefix for machine names.")]
+    [ValidateNotNullOrEmpty()]
+    [ValidateLength(3,7)]
+    [string]
+    $LabPrefix,
+
+    [Parameter(Mandatory,
+    ValueFromPipeline=$true, 
+    ValueFromPipelineByPropertyName=$true,
+    HelpMessage="Password for the local secretstore vault.")]
+    [ValidateNotNullOrEmpty()]
+    [securestring]
+    $SecretStorePassword
+)
+
+###################################################################################################
+
+function Write-LogFile {
+    param(
+        [Parameter(Mandatory = $true)]
+        [string] $Message
+    )
+ 
+    # Get the current date
+    $TimeStamp = Get-Date -Format o
+
+    # Add Content to the Log File
+    $Line = "$TimeStamp - $Message"
+    Add-content -Path $Logfile -Value $Line -ErrorAction SilentlyContinue
+    Write-Output $Line
+}
+
+
+Import-Module Microsoft.PowerShell.SecretManagement
+Import-Module Microsoft.PowerShell.SecretStore
+
+$LogFile = Join-Path $($env:Userprofile) "DJLog$(Get-Date -Format o | ForEach-Object { $_ -replace ":", "." }).txt"
+New-Item -Path $logFile -ItemType File
+
+# Check Windows 10 / 11 Operating system
+if (!([System.Environment]::OSVersion.Version.Major -match "10" -or [System.Environment]::OSVersion.Version.Major -match "11")) {
+    Write-LogFile "Requires Windows 10 or Windows 11."
+    exit
+}
+
+# Check if template
+$MetaDataHeaders = @{"Metadata"="true"}
+$vminfo = Invoke-RestMethod -Method GET -uri "http://169.254.169.254/metadata/instance?api-version=2018-10-01" -Headers $MetaDataHeaders
+
+if (!($vminfo.compute.vmScaleSetName -match "template")){
+    Write-Log "SaveDomainJoinValuesToSecretStore-TemplateVM script was not run on a template vm."
+    exit
+}
+
+# Password path
+$passwordPath = Join-Path $($env:Userprofile) SecretStore.vault.credential
+
+# if password file exists try to login with that
+if (!(Test-Path $passwordPath)) {
+
+    # Uses the DPAPI to encrypt the password
+    $SecretStorePassword | Export-CliXml $passwordPath 
+     
+}
+
+$pass = Import-CliXml $passwordPath
+# Check if store configuration exists
+$vault = Get-SecretVault
+
+if (!$vault) {
+
+    # if not create one
+    Set-SecretStoreConfiguration -Scope CurrentUser -Authentication Password -PasswordTimeout (60*60) -Interaction None -Password $pass -Confirm:$false
+    
+    Register-SecretVault -Name SecretStore -ModuleName Microsoft.PowerShell.SecretStore -DefaultVault
+ 
+}
+
+Unlock-SecretStore -Password $pass
+
+Set-Secret -Name LabId -Secret $LabPrefix
+
+# Copy down files into the Public documents folder
+# TODO set the correct final location
+Invoke-WebRequest -Uri https://raw.githubusercontent.com/Azure/LabServices/domainjoinv2/TemplateManagement/PowerShell/ActiveDirectoryJoinV2/Rename-StudentVM.ps1 -OutFile C:\Users\Public\Documents\Rename-StudentVM.ps1
```

Comment:
Nice.

I mentioned this previously, but you can use $env:PUBLIC\Documents\Rename-StudentVM.ps1